build_for_portal.py

```diff
@@ -611,65 +611,86 @@ def copy_file(
     with open(dest_file, "w") as f:
         f.write(content)
 
+# 7/26/24:
+# Corrected the scrub_file function.
+# Local files and URLs are now appropriately addressed individually.
+# URL includes of files within the openshift-docs repo now generate an error.
+# Files outside the `openshift-docs` repo are downloaded directly with error handling for connection issues.
 
 def scrub_file(info, book_src_dir, src_file, tag=None, cwd=None):
     """
     Scrubs a file and returns the cleaned file contents.
     """
     base_src_file = src_file.replace(info["src_dir"] + "/", "")
 
-    # added 1/Sep/2020
-    # to allow loading files like json and yaml from external sources, this
-    # procedure loads the file recognizing that it starts with http
-    # it then checks if it exists or not, and if it exists, returns the raw data
-    # data that it finds.
-    if base_src_file.startswith("https://raw.githubusercontent.com/openshift/"):
+    if base_src_file.startswith("https://raw.githubusercontent.com/"):
+        # Disallow URL inclusion from openshift-docs
+        if "openshift-docs" in base_src_file:
+            log.error(
+                "Inclusion of files within the openshift-docs repository by URL is not supported: %s",
+                base_src_file,
+            )
+            list_of_errors.append(
+                f"Inclusion of files within the openshift-docs repository by URL is not supported: {base_src_file}"
+            )
+            return ""  # Skip processing this file
+
+        # Allow includes only from specific organizations
+        if not any(org in base_src_file for org in ["openshift/", "redhatofficial/"]):
+            log.error(
+                "Inclusion of files from unauthorized repositories by URL is not supported: %s",
+                base_src_file,
+            )
+            list_of_errors.append(
+                f"Inclusion of files from unauthorized repositories by URL is not supported: {base_src_file}"
+            )
+            return ""  # Skip processing this file
+
         try:
             response = requests.get(base_src_file)
-            if response:
-                return response.text
+            if response.status_code == 200:
+                # Only apply scrubbing if the external file is AsciiDoc
```

Comment:
Why do we need to process external AsciiDoc files that are on the end of a raw github URL? I don't understand this piece.